spec/acceptance/basic_spec.rb

```diff
@@ -74,9 +74,20 @@ class { 'pulpcore':
     its(:exit_status) { is_expected.to eq 0 }
   end
 
-  describe command("PULP_SETTINGS=/etc/pulp/settings.py pulpcore-manager diffsettings") do
-    its(:stdout) { is_expected.to match(/^USE_NEW_WORKER_TYPE = True/) }
-    its(:exit_status) { is_expected.to eq 0 }
+  describe "Pulpcore Settings" do
+    let(:settings) { command("PULP_SETTINGS=/etc/pulp/settings.py pulpcore-manager diffsettings") }
```

Comment:
As written, this *works* and should give a more descriptive output in the case of test failures (I will break a test shortly to confirm).

That said, I want to make sure really I understand correctly how this behaves -- would the block argument here get evaluated in the `let` block, or does it later get evaluated three times when calling `settings.exit_status`, `settings.stdout`, etc, causing inefficient code execution?

What I'm trying to do may or may not still be possible with `describe command("PULP_SETTINGS=/etc/pulp/settings.py pulpcore-manager diffsettings") do` instead. I am not sure if `its(:stdout) "will a string work here?" do` would function as I intend it do.

From looking at a lot of rspec examples, `expect(...)` seems to be preferred to `it { is_expected.to ... }` which is why I decided to try this. That said, maybe it is proper rspec style in general but not right for serverspec specifically.

Reply:
I like the new grouping of things, and I think `let` should be evaluated once.

As for `expect` vs `is_expected.to` -- we use the later *everywhere*, so I'd prefer to keep that for consistency (and if change, change globally).

Reply:
What you can do is:
```ruby
describe 'Pulpcore settings' do
  subject { command("PULP_SETTINGS=/etc/pulp/settings.py pulpcore-manager diffsettings") }

  its(:exit_status) { is_expected.to eq 0 }
end
```

This is effectively the same as what was there before.

Reply:
I think I recall trying something like this in the past which failed:

```ruby
describe "Pulpcore settings" do
  subject { command("PULP_SETTINGS=/etc/pulp/settings.py pulpcore-manager diffsettings") }

  its(:exit_status) "exits without error" { is_expected.to eq 0 }
end
```

I will try again to confirm

Reply:
The issue was

```
Failure/Error: __send__(method, file)

SyntaxError:
  /home/wclark/Projects/puppet-pulpcore/spec/acceptance/basic_spec.rb:80: syntax error, unexpected string literal, expecting `end'
      its(:exit_status) "retrieves pulpcore settings" ...
```

So `it` has an optional string argument before the block argument, but `its(:symbol)` can receive only the block.